### ISSUE TITLE
Issues with g:syntastic_check_on_wq

### ISSUE THREAD

**Issue by baruchel** [scrooloose/syntastic#1479]
I have `let g:syntastic_check_on_wq = 0` in my ~/.vimrc and obviously syntastic doesn't check when I leave vim with the file being unmodified, but it still checks if I leave vim with :x after some modification.

I think this is an issue; the option has to discard the check in any case as long as the user has chosen to set the option.

**Comment by lcd047** [comment]
I can't reproduce the problem here, and the code is pretty straightforward.  If it doesn't work for you most likely there is something else going on.  What version of Vim are you running, on what OS?

**Comment by potmpark** [comment]
Not sure I'm having the same issue here. My syntastic related config in .vimrc:
```
let g:syntastic_mode_map = {
    \ "mode": "passive",
    \ "active_filetypes": [],
    \ "passive_filetypes": [] }
let g:syntastic_check_on_wq = 0
let g:syntastic_check_on_open = 0
let g:syntastic_auto_jump = 1
let g:syntastic_auto_loc_list = 1
let g:syntastic_enable_signs = 0
let g:syntastic_stl_format = '[%E{E:%fe #%e}%B{, }%W{W:%fw #%w}]'
nnoremap <silent><C-I> :SyntasticCheck<CR>
```
I want the checkers to check the file only when `<C-I>` is pressed, then popup location list and jump
to the first error, otherwise do nothing. But after `<C-I>` was pressed the first time, each time I use
`:w`, the location list would popup **having the same old error messages as before, even after I
corrected the error**. Using `:SyntasticReset` then would make everything normal as expected.
I'm using Vim 7.4 from the Ubuntu official repository on Ubuntu 14.10, writing racket using the default
racket checker, if that helps.

**Comment by lcd047** [comment]
@potmpark Please don't hijack somebody else's report, thank you.  Post a new issue instead.

**Comment by lcd047** [comment]
Ping.

**Comment by potmpark** [comment]
Adding a check for `g:syntastic_check_on_wq` in `syntastic/plugin/syntastic.vim` fixed this issue for me.
Change
```
function! s:BufWritePostHook() abort " {{{2
    call syntastic#log#debug(g:_SYNTASTIC_DEBUG_AUTOCOMMANDS,
            \ 'autocmd: BufWritePost, buffer ' . bufnr('') . ' = ' . string(bufname(str2nr(bufnr('')))))
    call s:UpdateErrors(1, [])
endfunction " }}}2
```
to
```
function! s:BufWritePostHook() abort " {{{2
    if g:syntastic_check_on_wq
        call syntastic#log#debug(g:_SYNTASTIC_DEBUG_AUTOCOMMANDS,
            \ 'autocmd: BufWritePost, buffer ' . bufnr('') . ' = ' . string(bufname(str2nr(bufnr('')))))
        call s:UpdateErrors(1, [])
    endif
endfunction " }}}2
```

**Comment by lcd047** [comment]
@potmpark That's great, but it still has nothing with either (1) the OP's problem, or (2) the problem you promised to report in a new issue, but didn't.

**Comment by lcd047** [comment]
A bug that might look like what you describe has been fixed in d26c7f7.  Please upgrade syntastic and try again.  Reopen this if the problem persists.
